### DISCUSSION TITLE
Deployment

### DISCUSSION THREAD

**Discussion by ShresthVaibhav** [PaiGramTeam/PaiGram#221] [General]
It's a great repo ....
Can I have a deployment guide ..... With each step .... And what to fill in env.... Thank for ur time

**Comment by Billyzou0741326** [comment]
Hi! Here's a quick attempt at translating the README docs to English (but with a few more steps). Bear with me if there's any confusion.

Requirements:

- Python 3.8+ (recommend Python 3.9 or 3.10)
- MySQL (for example `sudo apt install mysql-server-8.0`)
- Redis (for example `sudo apt install redis`)

1. Follow `poetry`'s [guide](https://python-poetry.org/docs/basic-usage/#using-your-virtual-environment) on setting up virtual environments. 
2. Install dependencies 

   ```
   poetry install
   playwright install chromium
   ```

   with optional dependency:

   ```
   poetry install --extras pyro
   ```
3. Configure your own `MySQL` and `Redis` instance on the server. By default, `MySQL` should be listening at `127.0.0.1:3306` and `Redis` listening at `127.0.0.1:6379`

   For MySQL, be sure to create a user and a database for this project:

   ```sql
   CREATE USER 'user'@'%' IDENTIFIED BY 'password';
   CREATE DATABASE 'paimon';
   GRANT ALL ON paimon.* TO 'user'@'%';
   ```

   The commands will create a user `user` with password `password` and a database named `paimon`. All privileges on the `paimon` database are granted to `user`.

   Redis doesn't need any configuration.

4. Configure `.env`:
   ```
   cp .env.example .env
   ```

   ```
   # debug toggle
   DEBUG=false
   
   # MySQL (required)
   DB_HOST=127.0.0.1
   DB_PORT=3306
   DB_USERNAME=user
   DB_PASSWORD="password"
   DB_DATABASE=paimon
   
   # Redis (required)
   REDIS_HOST=127.0.0.1
   REDIS_PORT=6379
   REDIS_DB=0
   
   # obtain token from https://t.me/BotFather use /newbot command to create a bot
   BOT_TOKEN="your_token_here"
   
   # bot administrators (it's ok to leave blank)
   ADMINS=[]
   
   # error reporting to dev group (optional)
   # ERROR_NOTIFICATION_CHAT_ID=chat_id
   
   # hoyo article publishing (optional)
   # CHANNELS=[{ "name": "", "chat_id": 1}]
   
   # group authentication (optional)
   # VERIFY_GROUPS=[]
   
   # logger config (optional)
   LOGGER_WIDTH=180
   LOGGER_LOG_PATH="logs"
   LOGGER_TIME_FORMAT="[%Y-%m-%d %X]"
   LOGGER_RENDER_KEYWORDS=["BOT"]
   # traceback options
   LOGGER_TRACEBACK_MAX_FRAMES=20
   LOGGER_LOCALS_MAX_DEPTH=0
   LOGGER_LOCALS_MAX_LENGTH=10
   LOGGER_LOCALS_MAX_STRING=80
   
   # mtp client (optional)
   # API_ID=12345
   # API_HASH="abcdefg"
   
   # ENKA_NETWORK_API (optional)
   # ENKA_NETWORK_API_AGENT=""
   
   # Web Server (optional)
   # Currently used for previewing templates, only enabled in dev environment
   # WEB_URL=http://localhost:8080/
   # WEB_HOST=localhost
   # WEB_PORT=8080
   ```

5. Initialize the database

   We use `alembic` to manage our database schema version:

   ```
   alembic upgrade head
   ```

   The command above will use the credentials configured in `.env` to connect to MySQL server and create the tables needed for this project

7. Run the project:

   ```
   python ./run.py
   ```

**Reply by Yuu-456** [reply]
FOR HEROKU 
can i just add dockerfile procfile and config in it ?

**Reply by Billyzou0741326** [reply]
Yes, feel free to define a Dockerfile with the required dependencies. Just make sure to provide a `.env` file when running the container

**Reply by Yuu-456** [reply]
> Yes, feel free to define a Dockerfile with the required dependencies. Just make sure to provide a `.env` file when running the container

thanks for the information

**Reply by luoshuijs** [reply]
> > Yes, feel free to define a Dockerfile with the required dependencies. Just make sure to provide a `.env` file when running the container
> 
> thanks for the information

@Yuu-456
Our team members have found that there is sensitive information leaked from your branch, please do not upload any fields of your own configuration file in the public repository. If you need to deploy to docker, we recommend deploying using environment variables rather than modifying the configuration file directly.

**Reply by luoshuijs** [reply]
@Yuu-456 Hopefully the reminder finds you well. We're currently prioritizing docker integration and internationalization. If you're willing to wait, we'll have these features ready in the next planned update.

**Reply by Yuu-456** [reply]
> > > Yes, feel free to define a Dockerfile with the required dependencies. Just make sure to provide a `.env` file when running the container
> > 
> > 
> > thanks for the information
> 
> @Yuu-456 Our team members have found that there is sensitive information leaked from your branch, please do not upload any fields of your own configuration file in the public repository. If you need to deploy to docker, we recommend deploying using environment variables rather than modifying the configuration file directly.

actually i know about the config.env files and i have only put some test values in it 
don't really worry it's all ok.
and i will private the main info anyway

**Reply by Yuu-456** [reply]
> @Yuu-456 Hopefully the reminder finds you well. We're currently prioritizing docker integration and internationalization. If you're willing to wait, we'll have these features ready in the next planned update.

well it would be really helpful if you add 

can you tell what would be the correct date when this update will come out ?

**Reply by luoshuijs** [reply]
> > @Yuu-456 Hopefully the reminder finds you well. We're currently prioritizing docker integration and internationalization. If you're willing to wait, we'll have these features ready in the next planned update.
> 
> well it would be really helpful if you add
> 
> can you tell what would be the correct date when this update will come out ?

Our maintenance team discussed this and found that things were not very simple, as some modules took a very long time to modify and validate when internationalizing.
For example, modules that rely on HTML need to adapt to string character length and prevent overflow because of the different text lengths in various languages. In the fastest case, it might take at least five months to release, but in the worst case, it would take at least a year or more.
